backend/src/main/java/com/swingevents/SwingEvents/EventController.java

```diff
@@ -1,37 +1,76 @@
 package com.swingevents.SwingEvents;
 
 import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
-import org.json.simple.parser.ParseException;
 import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.util.ResourceUtils;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.io.*;
-import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.List;
+import java.nio.file.Files;
+import java.util.*;
 
 
 @RestController
 public class EventController {
 
     @RequestMapping("/events")
-    public List<Event> seeAllEvents() throws IOException, ParseException {
-        List<Event> allEvents = new ArrayList<>();
-
-        JSONArray readEvents = readJSON();
-        allEvents.addAll(readEvents);
+    public List<Event> seeAllEvents() throws Exception {
+        List<Event> allEvents = readJSON();
         return allEvents;
     }
 
-    private static JSONArray readJSON() throws IOException, ParseException {
-        InputStream resource = new ClassPathResource(
-                "data/events.json").getInputStream();
-        JSONParser jsonParser = new JSONParser();
-        Object jsonObject = jsonParser.parse(
-                new InputStreamReader(resource, StandardCharsets.UTF_8));
-        return (JSONArray) jsonObject;
+
+    @RequestMapping("events/tags")
+    public Set<String> getAllTags(){
+        Set<String> tags = new HashSet<>();
+        try {
+            List<Event> events = readJSON();
+            for (Event event : events) {
+                String[] eventTags = event.getTags();
+                for (String tag:eventTags) {
+                    tags.add(tag);
+                }
```

Comment:
This may be simplified using Collection::addAll() method
See: https://www.geeksforgeeks.org/java-util-arraylist-addall-method-java/

Reply:
Done.